### PR TITLE
implement DataAPI

### PR DESCRIPTION
**Pull request by piever**
It is a draft as the package is not released yet (I'll make a Project.toml with the dependency when it is). See https://github.com/JuliaData/DataAPI.jl/issues/4 for the roadmap.

### CODE REVIEW

**Comment by piever** [comment]
Had forgotten about this one, I've added the DataAPI dependency in the Project file, as it was released in the meantime. It should be good to go now.

File: Project.toml [hunk 1]
```diff
@@ -2,8 +2,12 @@ name = "PooledArrays"
 uuid = "2dfb63ee-cc39-5dd5-95bd-886bf059d720"
 version = "0.5.2"
 
+[deps]
+DataAPI = "9a962f9c-6df0-11e9-0e5d-c546b8b5ee8a"
+
 [compat]
-julia = "0.7, 1"
+DataAPI = "1"
+julia = "1"
```
Comment (piever) [inline Project.toml:10]: Add to bump here as DataAPI requires julia 1.0 at least.

File: src/PooledArrays.jl [hunk 2]
```diff
@@ -137,6 +139,10 @@ PooledArray(t::Type, r::Type) = PooledArray(Array(t,0), r)
 ##
 ##############################################################################
 
+DataAPI.refarray(pa::PooledArray) = pa.refs
+DataAPI.refvalue(pa::PooledArray, i) = pa.pool[i]
```
Comment (nalimilan) [inline src/PooledArrays.jl]: Maybe restrict this to `Integer`?

Comment (piever) [reply src/PooledArrays.jl]: Good point, it should be an integer as it's used as an index to a vector, I've restricted the signature.

**Review by nalimilan — APPROVED** [review]

**Comment by nalimilan** [comment]
https://github.com/JuliaData/PooledArrays.jl/pull/31

**Pull request merged**